File: binding/src/charts/libqlegend.cpp
#include <QAbstractSeries>
#include <QBrush>
#include <QColor>
#include <QFont>
#include <QGraphicsWidget>
#include <QHideEvent>
#include <QLegend>
#include <QPainter>
#include <QPen>
#include <QShowEvent>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <type_traits>
#include <QStyleOptionGraphicsItem>
#include <QWidget>
#include <qlegend.h>
#include "libqlegend.h"
#include "libqlegend.hxx"

libqt_string QLegend_Tr(const char* s) {
	QString _ret = QLegend::tr(s);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QLegend_Paint(QLegend* self, QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget) {
	self->paint(painter, option, widget);
}

void QLegend_SetBrush(QLegend* self, const QBrush* brush) {
	self->setBrush(*brush);
}

QBrush* QLegend_Brush(const QLegend* self) {
	return new QBrush(self->brush());
}

void QLegend_SetColor(QLegend* self, QColor* color) {
	self->setColor(*color);
}

QColor* QLegend_Color(QLegend* self) {
	return new QColor(self->color());
}

void QLegend_SetPen(QLegend* self, const QPen* pen) {
	self->setPen(*pen);
}

QPen* QLegend_Pen(const QLegend* self) {
	return new QPen(self->pen());
}

void QLegend_SetBorderColor(QLegend* self, QColor* color) {
	self->setBorderColor(*color);
}

QColor* QLegend_BorderColor(QLegend* self) {
	return new QColor(self->borderColor());
}

void QLegend_SetFont(QLegend* self, const QFont* font) {
	self->setFont(*font);
}

QFont* QLegend_Font(const QLegend* self) {
	return new QFont(self->font());
}

void QLegend_SetLabelBrush(QLegend* self, const QBrush* brush) {
	self->setLabelBrush(*brush);
}

QBrush* QLegend_LabelBrush(const QLegend* self) {
	return new QBrush(self->labelBrush());
}

void QLegend_SetLabelColor(QLegend* self, QColor* color) {
	self->setLabelColor(*color);
}

QColor* QLegend_LabelColor(const QLegend* self) {
	return new QColor(self->labelColor());
}

void QLegend_SetAlignment(QLegend* self, int alignment) {
	self->setAlignment(static_cast<QFlags<Qt::AlignmentFlag>>(alignment));
}

int QLegend_Alignment(const QLegend* self) {
	return self->alignment();
}

void QLegend_DetachFromChart(QLegend* self) {
	self->detachFromChart();
}

void QLegend_AttachToChart(QLegend* self) {
	self->attachToChart();
}

bool QLegend_IsAttachedToChart(QLegend* self) {
	return self->isAttachedToChart();
}

void QLegend_SetBackgroundVisible(QLegend* self) {
	self->setBackgroundVisible();
}

bool QLegend_IsBackgroundVisible(const QLegend* self) {
	return self->isBackgroundVisible();
}

libqt_list QLegend_Markers(const QLegend* self) {
	auto _ret = self->markers();
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

bool QLegend_ReverseMarkers(QLegend* self) {
	return self->reverseMarkers();
}

void QLegend_SetReverseMarkers(QLegend* self) {
	self->setReverseMarkers();
}

bool QLegend_ShowToolTips(const QLegend* self) {
	return self->showToolTips();
}

void QLegend_SetShowToolTips(QLegend* self, bool show) {
	self->setShowToolTips(show);
}

bool QLegend_IsInteractive(const QLegend* self) {
	return self->isInteractive();
}

void QLegend_SetInteractive(QLegend* self, bool interactive) {
	self->setInteractive(interactive);
}

int QLegend_MarkerShape(const QLegend* self) {
	return self->markerShape();
}

void QLegend_SetMarkerShape(QLegend* self, int shape) {
	self->setMarkerShape(static_cast<QScatterSeries::MarkerShape>(shape));
}

void QLegend_BackgroundVisibleChanged(QLegend* self, bool visible) {
	self->backgroundVisibleChanged(visible);
}

void QLegend_Connect_BackgroundVisibleChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, bool) = reinterpret_cast<void (*)(QLegend*, bool)>(slot);
    QLegend::connect(self, &QLegend::backgroundVisibleChanged, [self, slotFunc](bool visible) {
	slotFunc(self, visible);
    });
}

void QLegend_ColorChanged(QLegend* self, QColor* color) {
	self->colorChanged(*color);
}

void QLegend_Connect_ColorChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, QColor*) = reinterpret_cast<void (*)(QLegend*, QColor*)>(slot);
    QLegend::connect(self, &QLegend::colorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QLegend_BorderColorChanged(QLegend* self, QColor* color) {
	self->borderColorChanged(*color);
}

void QLegend_Connect_BorderColorChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, QColor*) = reinterpret_cast<void (*)(QLegend*, QColor*)>(slot);
    QLegend::connect(self, &QLegend::borderColorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QLegend_FontChanged(QLegend* self, QFont* font) {
	self->fontChanged(*font);
}

void QLegend_Connect_FontChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, QFont*) = reinterpret_cast<void (*)(QLegend*, QFont*)>(slot);
    QLegend::connect(self, &QLegend::fontChanged, [self, slotFunc](QFont font) {
	slotFunc(self, font);
    });
}

void QLegend_LabelColorChanged(QLegend* self, QColor* color) {
	self->labelColorChanged(*color);
}

void QLegend_Connect_LabelColorChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, QColor*) = reinterpret_cast<void (*)(QLegend*, QColor*)>(slot);
    QLegend::connect(self, &QLegend::labelColorChanged, [self, slotFunc](QColor color) {
	slotFunc(self, color);
    });
}

void QLegend_ReverseMarkersChanged(QLegend* self, bool reverseMarkers) {
	self->reverseMarkersChanged(reverseMarkers);
}

void QLegend_Connect_ReverseMarkersChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, bool) = reinterpret_cast<void (*)(QLegend*, bool)>(slot);
    QLegend::connect(self, &QLegend::reverseMarkersChanged, [self, slotFunc](bool reverseMarkers) {
	slotFunc(self, reverseMarkers);
    });
}

void QLegend_ShowToolTipsChanged(QLegend* self, bool showToolTips) {
	self->showToolTipsChanged(showToolTips);
}

void QLegend_Connect_ShowToolTipsChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, bool) = reinterpret_cast<void (*)(QLegend*, bool)>(slot);
    QLegend::connect(self, &QLegend::showToolTipsChanged, [self, slotFunc](bool showToolTips) {
	slotFunc(self, showToolTips);
    });
}

void QLegend_MarkerShapeChanged(QLegend* self, int shape) {
	self->markerShapeChanged(static_cast<QScatterSeries::MarkerShape>(shape));
}

void QLegend_Connect_MarkerShapeChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, int) = reinterpret_cast<void (*)(QLegend*, int)>(slot);
    QLegend::connect(self, &QLegend::markerShapeChanged, [self, slotFunc](MarkerShape shape) {
	slotFunc(self, shape);
    });
}

void QLegend_AttachedToChartChanged(QLegend* self, bool attachedToChart) {
	self->attachedToChartChanged(attachedToChart);
}

void QLegend_Connect_AttachedToChartChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, bool) = reinterpret_cast<void (*)(QLegend*, bool)>(slot);
    QLegend::connect(self, &QLegend::attachedToChartChanged, [self, slotFunc](bool attachedToChart) {
	slotFunc(self, attachedToChart);
    });
}

void QLegend_InteractiveChanged(QLegend* self, bool interactive) {
	self->interactiveChanged(interactive);
}

void QLegend_Connect_InteractiveChanged(QLegend* self, intptr_t slot) {
    void (*slotFunc)(QLegend*, bool) = reinterpret_cast<void (*)(QLegend*, bool)>(slot);
    QLegend::connect(self, &QLegend::interactiveChanged, [self, slotFunc](bool interactive) {
	slotFunc(self, interactive);
    });
}

libqt_string QLegend_Tr2(const char* s, const char* c) {
	QString _ret = QLegend::tr(s, c);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

libqt_string QLegend_Tr3(const char* s, const char* c, int n) {
	QString _ret = QLegend::tr(s, c, n);
	QByteArray _b = _ret.toUtf8();
	libqt_string _str;
	_str.len = _b.length();
	_str.data = static_cast<const char*>(malloc(_str.len + 1));
	memcpy((void*)_str.data, _b.data(), _str.len);
	((char*)_str.data)[_str.len] = '\0';
	return _str;
}

void QLegend_SetBackgroundVisible1(QLegend* self, bool visible) {
	self->setBackgroundVisible(visible);
}

libqt_list QLegend_Markers1(const QLegend* self, QAbstractSeries* series) {
	auto _ret = self->markers(series);
	libqt_list _arr;
	_arr.len = _ret.length();
	_arr.data = malloc(_arr.len * sizeof(void*));
	void** _data = static_cast<void**>(_arr.data);
	for (int _i = 0; _i < _arr.len; ++_i) {
		_data[_i] = _ret[_i];
	}
	return _arr;
}

void QLegend_SetReverseMarkers1(QLegend* self, bool reverseMarkers) {
	self->setReverseMarkers(reverseMarkers);
}

// Base class handler implementation
void QLegend_QBasePaint(QLegend* self, QPainter* painter, const QStyleOptionGraphicsItem* option, QWidget* widget) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_Paint_IsBase(true);
	vqlegend->paint(painter, option, widget);
}
}

// Auxiliary method to allow providing re-implementation
void QLegend_OnPaint(QLegend* self, intptr_t slot) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_Paint_Callback(reinterpret_cast<VirtualQLegend::QLegend_Paint_Callback>(slot));
}
}

// Derived class handler implementation
void QLegend_HideEvent(QLegend* self, QHideEvent* event) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
	vqlegend->hideEvent(event);
	} else {
	self->QLegend::hideEvent(event);
}
}

// Base class handler implementation
void QLegend_QBaseHideEvent(QLegend* self, QHideEvent* event) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_HideEvent_IsBase(true);
	vqlegend->hideEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QLegend_OnHideEvent(QLegend* self, intptr_t slot) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_HideEvent_Callback(reinterpret_cast<VirtualQLegend::QLegend_HideEvent_Callback>(slot));
}
}

// Derived class handler implementation
void QLegend_ShowEvent(QLegend* self, QShowEvent* event) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
	vqlegend->showEvent(event);
	} else {
	self->QLegend::showEvent(event);
}
}

// Base class handler implementation
void QLegend_QBaseShowEvent(QLegend* self, QShowEvent* event) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_ShowEvent_IsBase(true);
	vqlegend->showEvent(event);
}
}

// Auxiliary method to allow providing re-implementation
void QLegend_OnShowEvent(QLegend* self, intptr_t slot) {
	auto* vqlegend = dynamic_cast<VirtualQLegend*>(self);
	if (vqlegend && vqlegend->isVirtualQLegend) {
vqlegend->setQLegend_ShowEvent_Callback(reinterpret_cast<VirtualQLegend::QLegend_ShowEvent_Callback>(slot));
}
}

void QLegend_Delete(QLegend* self) {
    delete self;
}

File: binding/include/charts/libqlegend.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qlegend.html
pub const qlegend = struct {

    pub fn Tr(s: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLegend_Tr(@ptrCast(s));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlegend.Tr: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Paint(self: ?*anyopaque, painter: ?*anyopaque, option: ?*anyopaque, widget: ?*anyopaque) void {
        qtc.QLegend_Paint(@ptrCast(self), @ptrCast(painter), @ptrCast(option), @ptrCast(widget));
    }

    /// Allows for overriding the related default method
    pub fn OnPaint(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque, ?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QLegend_OnPaint(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBasePaint(self: ?*anyopaque, painter: ?*anyopaque, option: ?*anyopaque, widget: ?*anyopaque) void {
        qtc.QLegend_QBasePaint(@ptrCast(self), @ptrCast(painter), @ptrCast(option), @ptrCast(widget));
    }

    pub fn SetBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QLegend_SetBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn Brush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QLegend_Brush(@ptrCast(self));
    }

    pub fn SetColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_SetColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn Color(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QLegend_Color(@ptrCast(self));
    }

    pub fn SetPen(self: ?*anyopaque, pen: ?*anyopaque) void {
        qtc.QLegend_SetPen(@ptrCast(self), @ptrCast(pen));
    }

    pub fn Pen(self: ?*anyopaque, ) QtC.QPen {
        return qtc.QLegend_Pen(@ptrCast(self));
    }

    pub fn SetBorderColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_SetBorderColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn BorderColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QLegend_BorderColor(@ptrCast(self));
    }

    pub fn SetFont(self: ?*anyopaque, font: ?*anyopaque) void {
        qtc.QLegend_SetFont(@ptrCast(self), @ptrCast(font));
    }

    pub fn Font(self: ?*anyopaque, ) QtC.QFont {
        return qtc.QLegend_Font(@ptrCast(self));
    }

    pub fn SetLabelBrush(self: ?*anyopaque, brush: ?*anyopaque) void {
        qtc.QLegend_SetLabelBrush(@ptrCast(self), @ptrCast(brush));
    }

    pub fn LabelBrush(self: ?*anyopaque, ) QtC.QBrush {
        return qtc.QLegend_LabelBrush(@ptrCast(self));
    }

    pub fn SetLabelColor(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_SetLabelColor(@ptrCast(self), @ptrCast(color));
    }

    pub fn LabelColor(self: ?*anyopaque, ) QtC.QColor {
        return qtc.QLegend_LabelColor(@ptrCast(self));
    }

    pub fn SetAlignment(self: ?*anyopaque, alignment: i32) void {
        qtc.QLegend_SetAlignment(@ptrCast(self), @intCast(alignment));
    }

    pub fn Alignment(self: ?*anyopaque, ) i32 {
        return qtc.QLegend_Alignment(@ptrCast(self));
    }

    pub fn DetachFromChart(self: ?*anyopaque, ) void {
        qtc.QLegend_DetachFromChart(@ptrCast(self));
    }

    pub fn AttachToChart(self: ?*anyopaque, ) void {
        qtc.QLegend_AttachToChart(@ptrCast(self));
    }

    pub fn IsAttachedToChart(self: ?*anyopaque, ) bool {
        return qtc.QLegend_IsAttachedToChart(@ptrCast(self));
    }

    pub fn SetBackgroundVisible(self: ?*anyopaque, ) void {
        qtc.QLegend_SetBackgroundVisible(@ptrCast(self));
    }

    pub fn IsBackgroundVisible(self: ?*anyopaque, ) bool {
        return qtc.QLegend_IsBackgroundVisible(@ptrCast(self));
    }

    pub fn Markers(self: ?*anyopaque, ) []const u8 {
        return qtc.QLegend_Markers(@ptrCast(self));
    }

    pub fn ReverseMarkers(self: ?*anyopaque, ) bool {
        return qtc.QLegend_ReverseMarkers(@ptrCast(self));
    }

    pub fn SetReverseMarkers(self: ?*anyopaque, ) void {
        qtc.QLegend_SetReverseMarkers(@ptrCast(self));
    }

    pub fn ShowToolTips(self: ?*anyopaque, ) bool {
        return qtc.QLegend_ShowToolTips(@ptrCast(self));
    }

    pub fn SetShowToolTips(self: ?*anyopaque, show: bool) void {
        qtc.QLegend_SetShowToolTips(@ptrCast(self), show);
    }

    pub fn IsInteractive(self: ?*anyopaque, ) bool {
        return qtc.QLegend_IsInteractive(@ptrCast(self));
    }

    pub fn SetInteractive(self: ?*anyopaque, interactive: bool) void {
        qtc.QLegend_SetInteractive(@ptrCast(self), interactive);
    }

    pub fn MarkerShape(self: ?*anyopaque, ) i32 {
        return qtc.QLegend_MarkerShape(@ptrCast(self));
    }

    pub fn SetMarkerShape(self: ?*anyopaque, shape: i32) void {
        qtc.QLegend_SetMarkerShape(@ptrCast(self), @intCast(shape));
    }

    pub fn HideEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QLegend_HideEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnHideEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QLegend_OnHideEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseHideEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QLegend_QBaseHideEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn ShowEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QLegend_ShowEvent(@ptrCast(self), @ptrCast(event));
    }

    /// Allows for overriding the related default method
    pub fn OnShowEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QLegend_OnShowEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseShowEvent(self: ?*anyopaque, event: ?*anyopaque) void {
        qtc.QLegend_QBaseShowEvent(@ptrCast(self), @ptrCast(event));
    }

    pub fn BackgroundVisibleChanged(self: ?*anyopaque, visible: bool) void {
        qtc.QLegend_BackgroundVisibleChanged(@ptrCast(self), visible);
    }

    pub fn OnBackgroundVisibleChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QLegend_Connect_BackgroundVisibleChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_ColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QLegend_Connect_ColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn BorderColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_BorderColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnBorderColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QLegend_Connect_BorderColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn FontChanged(self: ?*anyopaque, font: QtC.QFont) void {
        qtc.QLegend_FontChanged(@ptrCast(self), @ptrCast(font));
    }

    pub fn OnFontChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QFont) callconv(.c) void) void {
        qtc.QLegend_Connect_FontChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn LabelColorChanged(self: ?*anyopaque, color: QtC.QColor) void {
        qtc.QLegend_LabelColorChanged(@ptrCast(self), @ptrCast(color));
    }

    pub fn OnLabelColorChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, QtC.QColor) callconv(.c) void) void {
        qtc.QLegend_Connect_LabelColorChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ReverseMarkersChanged(self: ?*anyopaque, reverseMarkers: bool) void {
        qtc.QLegend_ReverseMarkersChanged(@ptrCast(self), reverseMarkers);
    }

    pub fn OnReverseMarkersChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QLegend_Connect_ReverseMarkersChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn ShowToolTipsChanged(self: ?*anyopaque, showToolTips: bool) void {
        qtc.QLegend_ShowToolTipsChanged(@ptrCast(self), showToolTips);
    }

    pub fn OnShowToolTipsChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QLegend_Connect_ShowToolTipsChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn MarkerShapeChanged(self: ?*anyopaque, shape: i32) void {
        qtc.QLegend_MarkerShapeChanged(@ptrCast(self), @intCast(shape));
    }

    pub fn OnMarkerShapeChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QLegend_Connect_MarkerShapeChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn AttachedToChartChanged(self: ?*anyopaque, attachedToChart: bool) void {
        qtc.QLegend_AttachedToChartChanged(@ptrCast(self), attachedToChart);
    }

    pub fn OnAttachedToChartChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QLegend_Connect_AttachedToChartChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn InteractiveChanged(self: ?*anyopaque, interactive: bool) void {
        qtc.QLegend_InteractiveChanged(@ptrCast(self), interactive);
    }

    pub fn OnInteractiveChanged(self: ?*anyopaque, callback: *const fn (?*anyopaque, bool) callconv(.c) void) void {
        qtc.QLegend_Connect_InteractiveChanged(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    pub fn Tr2(s: []const u8, c: []const u8, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLegend_Tr2(@ptrCast(s), @ptrCast(c));
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlegend.Tr2: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn Tr3(s: []const u8, c: []const u8, n: i32, allocator: std.mem.Allocator) []const u8 {
        const _str = qtc.QLegend_Tr3(@ptrCast(s), @ptrCast(c), n);
defer qtc.libqt_string_free(&_str);
const _ret = allocator.alloc(u8, _str.len) catch @panic("qlegend.Tr3: Memory allocation failed");
@memcpy(_ret, _str.data[0.._str.len]);
return  _ret;
    }

    pub fn SetBackgroundVisible1(self: ?*anyopaque, visible: bool) void {
        qtc.QLegend_SetBackgroundVisible1(@ptrCast(self), visible);
    }

    pub fn Markers1(self: ?*anyopaque, series: ?*anyopaque) []const u8 {
        return qtc.QLegend_Markers1(@ptrCast(self), @ptrCast(series));
    }

    pub fn SetReverseMarkers1(self: ?*anyopaque, reverseMarkers: bool) void {
        qtc.QLegend_SetReverseMarkers1(@ptrCast(self), reverseMarkers);
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QLegend_Delete(@ptrCast(self));
    }
};

/// https://doc.qt.io/qt-6/legend.html#types
pub const enums = struct {
    pub const MarkerShape = enum {
        pub const MarkerShapeDefault: i32 = 0;
        pub const MarkerShapeRectangle: i32 = 1;
        pub const MarkerShapeCircle: i32 = 2;
        pub const MarkerShapeFromSeries: i32 = 3;
        pub const MarkerShapeRotatedRectangle: i32 = 4;
        pub const MarkerShapeTriangle: i32 = 5;
        pub const MarkerShapeStar: i32 = 6;
        pub const MarkerShapePentagon: i32 = 7;
    };

};
